src/services/sync/strategies/LDAPSyncer.js

```diff
@@ -38,212 +40,87 @@ class LDAPSyncer extends Syncer {
 	async steps() {
 		await super.steps();
 		await this.syncQueue.getChannel();
-		await this.attemptRun();
-		const schools = await this.getSchools();
-		const userPromises = [];
-		for (const school of schools) {
-			userPromises.push(this.getUserData(school));
-		}
-		await Promise.all(userPromises);
+		const schools = await this.processLdapSchools();
+		await Promise.all(schools.map((school) => this.processLdapUsers(school)));
+		await Promise.all(schools.map((school) => this.processLdapClasses(school)));
+	}
 
-		const schoolPromises = [];
-		for (const school of schools) {
-			schoolPromises.push(this.getClassData(school));
-		}
-		await Promise.all(schoolPromises);
+	async processLdapSchools() {
+		const ldapSchools = await this.ldapService.getSchools(this.system.ldapConfig);
+		this.stats.schools += ldapSchools.length;
+		const years = await SchoolRepo.getYears();
+		const currentYear = new SchoolYearFacade(years).defaultYear;
```

Comment:
Are current school and federal state not part of the school? The default year must not match the schools year.

Reply:
I am not sure, but i think this is only used for creating new schools.

Reply:
federal state and current year not part of the school in ldap.

Reply:
If it is used for other stuff like create, than it must fetch from the school for the selected year for each school.